extension/src/components/Popup.tsx

```diff
@@ -3,41 +3,31 @@ import {
     AppBar,
     Card,
     CardContent,
-    CardHeader,
     CircularProgress,
     CssBaseline,
-    Link,
     Toolbar,
     Typography
 } from '@mui/material';
 import SettingsIcon from '@mui/icons-material/Settings';
-import wretch from 'wretch';
-import { PopupToBackgroundMsg, BackgroundToPopupMsg, MsgType, MLISResponse } from "../extension/types";
-
-// TODO: Issue #14: Change url and endpoint after MLIS is ready
-const apiRoot: string = "http://localhost:8000";
-const summarizationEndpoint: string = "/summary";
+import { MLISRequest, MLISResponse } from "../extension/types";
 
 function Popup() {
     const [isLoading, setIsLoading] = useState(true);
     const [content, setContent] = useState('');
     const [articleTitle, setArticleTitle] = useState('');
 
     useEffect(() => {
-        const msg: PopupToBackgroundMsg = {
-            type: MsgType.PopUpInit
-        };
-        chrome.runtime.sendMessage(msg, (response: BackgroundToPopupMsg) => {
-            if (response && response.textToSummarize) {
-                wretch(apiRoot + summarizationEndpoint)
-                    .options({ mode: "cors" })
-                    .post({ text: response.textToSummarize })
-                    .json((mlisResponse: MLISResponse) => {
-                        setArticleTitle(response.articleTitle);
-                        setContent(mlisResponse.text);
-                        setIsLoading(false);
-                    });
-            }
+        chrome.tabs.query({active: true, lastFocusedWindow: true}, tabs => {
+            const request: MLISRequest = {
+                url: tabs[0].url,
```

Comment:
Just checked chrome developer guide that chrome.tabs.query({active: true, lastFocusedWindow: true} always returns a list of length one. Would it be clearer to do (not sure if it would work syntactically with lambda)
```
chrome.tabs.query({active: true, lastFocusedWindow: true}, [tab] => {
            const request: MLISRequest = {
                url: tab.url,
```

Reference: https://developer.chrome.com/docs/extensions/reference/tabs/ 
"Get the current tab" section

Reply:
Thanks for looking it up 👍
It would make sense to use an array of size 1 instead of arbitrary size.